executor/sort.go

```diff
@@ -307,7 +307,7 @@ func (e *TopNExec) Next(ctx context.Context, chk *chunk.Chunk) error {
 	}
 	chk.Reset()
 	if !e.fetched {
-		e.totalLimit = int(e.limit.Offset + e.limit.Count)
+		e.totalLimit = e.limit.Offset + e.limit.Count
```

Comment:
Will it overflow uint64?

Reply:
Yes, but it seems to happen with little probability in a real scenario.